src/pods/toolbar/components/export-button/export-button.business.ts

```diff
@@ -28,6 +28,78 @@ export const calculateTableEndYPosition = (table: TableVm): number => {
   return table.y + tableHeight;
 };
 
+export const isTableOverlap = (table: TableVm, tables: TableVm[]): boolean => {
+  if (tables.length === 0) {
+    return false;
+  }
+
+  const expandedFields = expandTableFields(table.fields);
+  const tableEndXPosition = table.x + TABLE_CONST.TABLE_WIDTH;
+  const tableEndYPosition = calculateTableEndYPosition({
+    ...table,
+    fields: expandedFields,
+  });
+
+  return tables.some(t => {
+    if (t.id === table.id) {
+      return false;
+    }
+
+    const tExpandedFields = expandTableFields(t.fields);
+    const tEndXPosition = t.x + TABLE_CONST.TABLE_WIDTH;
+    const tEndYPosition = calculateTableEndYPosition({
+      ...t,
+      fields: tExpandedFields,
+    });
+
+    const overlapInX = table.x <= tEndXPosition && tableEndXPosition >= t.x;
+    const overlapInY = table.y <= tEndYPosition && tableEndYPosition >= t.y;
+
+    return overlapInX && overlapInY;
```

Comment:
following that approach it could get simpler:

```ts
return doTablesOverlap(tableA, tableB);

Reply:
done